crates/core/src/network/networking.rs

```diff
@@ -181,16 +183,15 @@ impl<T: TransportConnect> NetworkSender<NoConnection> for Networking<T> {
                         | e @ NetworkError::ConnectError(_)
                         | e @ NetworkError::ConnectionClosed,
                     ) => {
-                        info!(
-                            "Connection to node {} failed with {}, next retry is attempt {}/{}",
-                            msg.peer(),
-                            e,
-                            attempts + 1,
-                            self.options
-                                .connect_retry_policy
-                                .max_attempts()
-                                .unwrap_or(NonZeroUsize::MAX), // max_attempts() be Some at this point
-                        );
+                        if next_attempt >= max_attempts {
+                            trace!(
+                                "Connection to node {} failed with {}, next retry is attempt {}/{}",
+                                msg.peer(),
+                                e,
+                                next_attempt,
+                                max_attempts,
+                            );
+                        }
```

Comment:
Should this only be logged on the final attempt? If yes, then I guess we don't need to log `next retry is attempt {}/{}`.

Reply:
Right, let me remove this.